lib/phoenix/router/route.ex

```diff
@@ -48,7 +48,7 @@ defmodule Phoenix.Router.Route do
   """
   @spec build(non_neg_integer, :match | :forward, atom, String.t, String.t | nil, atom, atom, atom | nil, list(atom), map, map, map, boolean, boolean) :: t
   def build(line, kind, verb, path, host, plug, plug_opts, helper, pipe_through, private, assigns, metadata, trailing_slash?, warn_on_verify?)
-      when is_atom(verb) and (is_binary(host) or is_nil(host)) and
+      when is_atom(verb) and (is_binary(host) or is_nil(host) or is_list(host)) and
```

Comment:
The host is only created in `Scope.push`. So my suggestion is to normalize the host to a list as soon as possible in there, by calling `List.wrap`. Everywhere else, the host is a list. Then you can rename the `host` field to `hosts` both there and here.